mater/lib/src/multicodec.rs

```diff
@@ -35,3 +38,8 @@ where
     Multihash::wrap(H::CODE, &hashed_bytes)
         .expect("the digest should be valid (enforced by the type system)")
 }
+
+// Returns Some(data) if the CID is an identity. If not, None is returned.
+pub fn is_identity<const S: usize>(cid: &CidGeneric<S>) -> Option<&[u8]> {
+    (cid.hash().code() == IDENTITY_CODE).then_some(cid.hash().digest())
```

Comment:
The contents of `.then_some` are always evaluated even if the bool is false, this alternative only gets the digest iff the bool is true
```suggestion
    (cid.hash().code() == IDENTITY_CODE).then(|| cid.hash().digest())
```